### ISSUE TITLE
V4: Unable to override style properties in @shopify/flashlist 

### ISSUE THREAD

**Issue by xinha-sh** [nativewind/nativewind#660]
**Describe the bug**
``ListHeaderComponentClassName`` doesn't work. Tried using ``remapProps`` or ``cssInterop`` but still no success.

**To Reproduce**
1. Follow instruction to install Flashlist: [https://github.com/Shopify/flash-list](https://github.com/Shopify/flash-list)
2. Use in App.tsx
```tsx
import React from "react";
import { View, Text } from "react-native";
import { FlashList } from "@shopify/flash-list";

const DATA = [
  {
    title: "First Item",
  },
  {
    title: "Second Item",
  },
];

const MyList = () => {
  return (
    <FlashList
      data={DATA}
      renderItem={({ item }) => <Text>{item.title}</Text>}
      estimatedItemSize={200}
      ListHeaderComponentClassName="px-24" // <--- This will show Property ListHeaderComponentClassName does not exist on type
    />
  );
};
```

**Expected behavior**
Work similar to FlatList or remapProps / cssInterop to work with Flashlist

**Screenshots**
![Screenshot 2023-11-21 at 5 21 03 PM](https://github.com/marklawlor/nativewind/assets/107257599/3297e562-d0a6-4b42-82c0-bae755f3a820)

**Additional context**
I checked the expected type for both FlatList and FlashList and found that FlashList is expecting value that satisfies
[link](https://shopify.github.io/flash-list/docs/usage#listheadercomponentstyle)
```
ListHeaderComponentStyle?: StyleProp<ViewStyle>;
```
instead of [FlatList's](https://reactnative.dev/docs/flatlist#listheadercomponentstyle)
```
ListHeaderComponentStyle?: ViewStyle;
```

**Comment by johnf** [comment]
This might be fixed by #690 if you're using 4.0.16

**Comment by danstepanov** [comment]
We are closing all pre-v4.1 issues now that v4.1 is released. If the problem is still happening with the latest v4.1, [please open a new issue](https://github.com/nativewind/nativewind/blob/next/contributing.md#opening-an-issue) and reference this one in the description.
